server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -939,14 +949,46 @@ public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext
                 );
             }
 
-            throw new IllegalStateException("unknown field data operation [" + operation.name() + "]");
+            if (operation != FielddataOperation.SCRIPT) {
+                throw new IllegalStateException("unknown field data operation [" + operation.name() + "]");
+            }
+            if (isSyntheticSource && isStored()) {
```

Comment:
Same question here

Reply:
This one has a different answer! This *will* be possible in a follow up but isn't yet. I'll throw a helpful exception here too.

Reply:
This one has a different answer! This *will* be possible in a follow up but isn't yet. I'll throw a helpful exception here too.

Reply:
This one has a different answer! This *will* be possible in a follow up but isn't yet. I'll throw a helpful exception here too.